internal/ansible/ansible_manager_test.go

```diff
@@ -0,0 +1,261 @@
+				Fail("cannot crete message content")
+			}
+			message := pb.Data{
+				MessageId: messageID,
+				Metadata:  m,
+				Content:   msgContentByte,
+			}
+
+			//when
+			err = ansibleManager.HandlePlaybook(playbookCmd, &message, timeout)
+			Expect(err).ToNot(HaveOccurred())
+			//then
+			Expect(client.latestData).ToNot(BeNil())
+			Expect(client.latestData.Directive).To(Equal(returnUrl))
+
+		})
+		It("Test ansible playbook ", func() {
+
+			//given
+			playbookFilename := "examples/test_playbook_2.yml"
+			playbookCmd = &playbook.AnsiblePlaybookCmd{
+				Playbooks:         []string{playbookFilename},
+				ConnectionOptions: ansiblePlaybookConnectionOptions,
+				Options:           ansiblePlaybookOptions,
+			}
+
+			ansibleManager, err := ansible.NewAnsibleManager(&client, configDir)
+			Expect(err).ToNot(HaveOccurred())
+
+			playbookContent, err := os.ReadFile(playbookFilename)
+			if err != nil {
+				Fail("cannot read playbook test file: " + playbookFilename)
+			}
+
+			m := map[string]string{"ansible-playbook": "true", "crc_dispatcher_correlation_id": "d4ae95cf-71fd-4386-8dbf-2bce933ce713", "response_interval": "250", "return_url": returnUrl}
+			msgContent := map[string]string{"ansible_playbook": string(playbookContent)}
+			msgContentByte, err := json.Marshal(msgContent)
+			if err != nil {
+				Fail("cannot crete message content")
+			}
+			message := pb.Data{
+				MessageId: messageID,
+				Metadata:  m,
+				Content:   msgContentByte,
+			}
+
+			//when
+			err = ansibleManager.HandlePlaybook(playbookCmd, &message, timeout)
+			Expect(err).ToNot(HaveOccurred())
+			Expect(client.latestData.Directive).To(Equal(returnUrl))
+		})
+		It("Execute slow ansible playbook", func() {
+
+			timeout = 1 * time.Millisecond
+			//given
+			playbookFilename := "examples/test_playbook_timeout.yml"
+			playbookCmd = &playbook.AnsiblePlaybookCmd{
+				Playbooks:         []string{playbookFilename},
+				ConnectionOptions: ansiblePlaybookConnectionOptions,
+				Options:           ansiblePlaybookOptions,
+			}
+
+			ansibleManager, err := ansible.NewAnsibleManager(&client, configDir)
+			Expect(err).ToNot(HaveOccurred())
+
+			playbookContent, err := os.ReadFile(playbookFilename)
+			if err != nil {
+				Fail("cannot read playbook test file: " + playbookFilename)
+			}
+
+			m := map[string]string{"ansible-playbook": "true", "crc_dispatcher_correlation_id": "d4ae95cf-71fd-4386-8dbf-2bce933ce713", "response_interval": "250", "return_url": returnUrl}
+			msgContent := map[string]string{"ansible_playbook": string(playbookContent)}
+			msgContentByte, err := json.Marshal(msgContent)
+			if err != nil {
+				Fail("cannot crete message content")
+			}
+			message := pb.Data{
+				MessageId: messageID,
+				Metadata:  m,
+				Content:   msgContentByte,
+			}
+
+			//when
+			err = ansibleManager.HandlePlaybook(playbookCmd, &message, timeout)
+			Expect(err).To(HaveOccurred())
+			Expect(client.latestData).To(BeNil())
+		})
+	})
+	BeforeEach(func() {
```

Comment:
I'm worried here, this BeforeEach/AfterEach are defined twice, maybe this should be inside of Context("on-restart"), no?